climada/engine/test/test_impact.py

```diff
@@ -69,6 +69,123 @@ def test_from_eih_pass(self):
             np.stack([exp.gdf.latitude.values, exp.gdf.longitude.values], axis=1)
             )
 
+    def test_concat_pass(self):
+        """test concatenation of impacts"""
+
+        def test_haztype_unit_crs_attrs(self,imp1,imp2,imp3,imp3_3,imp3_2):
+            with self.assertRaises(ValueError):
+                Impact.concat([imp1, imp2, imp3_3])
+            with self.assertRaises(ValueError):
+                Impact.concat([imp1, imp2, imp3_2])
+            with self.assertRaises(ValueError):
+                imp3.unit = ''
+                Impact.concat([imp1, imp2, imp3])
+            with self.assertRaises(ValueError):
+                imp3.unit = 'OTHER'
+                Impact.concat([imp1, imp2, imp3])
+            imp3.unit = imp2.unit
+            with self.assertRaises(ValueError):
+                imp3.crs = ''
+                Impact.concat([imp1, imp2, imp3])
+            with self.assertRaises(ValueError):
+                imp3.crs = 'OTHER'
+                Impact.concat([imp1, imp2, imp3])
+            imp3.crs = imp2.crs
+            with self.assertRaises(ValueError):
+                imp3.new_attr = 'test_attr'
+                Impact.concat([imp1, imp2, imp3])
+            delattr(imp3, "new_attr")
```

Comment:
This test is very very cryptic. What are the errors that raised and for what?

You should test the specific error raised (check the error message). Also, the code is not readable, so please choose better names, add some comments, and maybe breakup in smaller bits.

Reply:
Check for specific errors like this:
```python
with self.assertRaises(SomeError) as cm:
    raise_error()
self.assertIn("part of a specific error message I was expecting", str(cm.exception))
```